core/azure-core/src/main/java/com/azure/core/util/polling/Poller.java

```diff
@@ -0,0 +1,285 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.util.polling;
+
+import reactor.core.Disposable;
+
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.time.Duration;
+import java.util.Objects;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+/**
+ * This class provides implementation of long running operations. The poller starts polling <b>automatically</b> when Poller instance is created.
+ * It uses {@link Flux} from reactive programming model to achieve auto polling.
+ * It has function for usual operation of a poller. For example listen/observe poll responses, enable/disable auto polling,
+ * manual polling, wait for polling to complete and get status of current polling.
+ *
+ * {@link Mono} from poll operation should never return Mono.error(). If any unexpected
+ * scenario happens in poll operation, it should catch it and return a valid PollResponse.
+ * This is because poller does not know what to do in case on Mono.error().
+ * and we will continue to poll in this scenario
+ *
+ * <p><strong>Code Samples</strong></p>
+ *
+ * <p><strong>Instantiating and Subscribing to Poll Response</strong></p>
+ *
+ * {@codesnippet com.azure.core.util.polling.poller.instantiationAndSubscribe}
+ *
+ * <p><strong>Wait/Block for Polling to complete</strong></p>
+ * {@codesnippet com.azure.core.util.polling.poller.block}
+ *
+ * <p><strong>Disable auto polling and polling manually</strong></p>
+ * {@codesnippet com.azure.core.util.polling.poller.poll}
+ *
+ * <p><strong>Implementation of Long Running Operations</strong></p>
+ *
+ * @param <T> Type of poll response value
+ *
+ * @see PollResponse
+ * @see com.azure.core.util.polling.PollResponse.OperationStatus
+ */
```

Comment:
Need to include discussion of auto-polling, including what it means, how to disable it, what it means to disable it, why you might want to do this, etc.